### ISSUE TITLE
Staking reddcoin but not receiving any rewards?

### ISSUE THREAD

**Issue by toddbeckmann** [reddcoin-project/reddcoin-3.10#171]
My wallet is set for staking but, I have not received any transactions since upgrading to the latest version of Reddcoin core wallet. I had issues with the upgrade and went back to the previous version for a few days in December. I did receive two interest payments during that short period. I then reinstalled the latest version did a wallet sync and came up with what I believe to be an accurate value for the coins held. 
Please review the attached files (with my comments). I need help figuring out the problem with staking.
![Reddcoin screen Capture 6 2-23-2021](https://user-images.githubusercontent.com/41970872/108912338-8349f980-75f6-11eb-9400-1606748cac1c.jpg)
![Reddcoin screen Capture 7 2-23-2021](https://user-images.githubusercontent.com/41970872/108912342-83e29000-75f6-11eb-8d9a-3ab666ca5511.jpg)
![Reddcoin screen Capture 8 2-23-2021](https://user-images.githubusercontent.com/41970872/108912345-83e29000-75f6-11eb-857b-1aa7b9dd6fcc.jpg)
![Reddcoin screen Capture 9 2-23-2021](https://user-images.githubusercontent.com/41970872/108912348-8513bd00-75f6-11eb-9222-6630b0bdb101.jpg)
![Reddcoin screen Capture 10 2-23-2021](https://user-images.githubusercontent.com/41970872/108912351-8644ea00-75f6-11eb-8583-77297de16f27.jpg)
![Reddcoin screen Capture 11 2-23-2021](https://user-images.githubusercontent.com/41970872/108912352-8644ea00-75f6-11eb-8fe6-af67f2ed6ad4.jpg)
![Reddcoin screen Capture 12 2-23-2021](https://user-images.githubusercontent.com/41970872/108912355-86dd8080-75f6-11eb-826b-2aac9bf8340a.jpg)
![Reddcoin screen Capture 13 2-23-2021](https://user-images.githubusercontent.com/41970872/108912357-86dd8080-75f6-11eb-8680-4ff48cace9cc.jpg)
![Reddcoin screen Capture 13b 2-23-2021](https://user-images.githubusercontent.com/41970872/108912359-87761700-75f6-11eb-94a4-13907016ade2.jpg)
![Reddcoin screen Capture 13c 2-23-2021](https://user-images.githubusercontent.com/41970872/108912360-880ead80-75f6-11eb-97fc-ce14a1322f33.jpg)
![Reddcoin screen Capture 14 2-23-2021](https://user-images.githubusercontent.com/41970872/108912364-88a74400-75f6-11eb-90ea-f1de83797848.jpg)
![Reddcoin screen Capture 15 2-23-2021](https://user-images.githubusercontent.com/41970872/108912366-893fda80-75f6-11eb-9d6d-dfc0f49afedd.jpg)
![Reddcoin screen Capture 1 2-23-2021 2-54-26 PM](https://user-images.githubusercontent.com/41970872/108912370-893fda80-75f6-11eb-972f-e4bcd685c8d8.jpg)
![Reddcoin screen Capture 2 2-23-2021 2-54-26 PM](https://user-images.githubusercontent.com/41970872/108912371-89d87100-75f6-11eb-917d-220f9902fe07.jpg)
![Reddcoin screen Capture 3 2-23-2021](https://user-images.githubusercontent.com/41970872/108912372-89d87100-75f6-11eb-9f90-568dfde9e526.jpg)
![Reddcoin screen Capture 4 2-23-2021](https://user-images.githubusercontent.com/41970872/108912374-8b099e00-75f6-11eb-9fd4-343b580e8e0a.jpg)
![Reddcoin screen Capture 5 2-23-2021](https://user-images.githubusercontent.com/41970872/108912376-8ba23480-75f6-11eb-9ba7-90ef91ccba73.jpg)

**Comment by reddink** [comment]
Thanks for the detailed information.
If your wallet balance in the new wallet matches, you should be in good shape.
As you noticed, there is an issue with the balance displayed for the RPC listaccounts command

The correct value, is shown on the Overview Tab.
(You can also see the correct amounts for addresses if you enable "coin control" in the settings, and then in the "Send" Tab, navigate to the "Inputs" button 

For staking:
(I notice from the Overview Tab that you also got a stake rewards January 10.)
There are a few factors in play that determine the frequency of rewards.
Firstly, the wallet will earn 5% per annum
Network weight (How many other people are staking) will influence when stakes are rewarded.
Think of it as a competition:
 A block is only generated every minute. Every user is competing to generate a block, and as more users stake, the difficulty increases making it harder to generate a block.

Your wallet hasnt staked since mid 2019, so coins have earned a certain level of coinage, and weight which accounts for the large stake on December 10.

For staking consistency, you should leave your wallet online and staking enabled.

Staking behaviour is the same for the 3.10.3 wallet

You can hover your mouse pointer over the green icon in the status bar, to display the approx time until next stake.

In summary,
You balance looks correct (according to Overview Tab)
We have an code issue to sort out in the display of listaccounts RPC call
Your wallet is staking, (has staked 3 times since bringing wallet online)
Use the green staking status icon to display the approx time until next stake

**Comment by toddbeckmann** [comment]
John,

 

Thanks for the informative response. 

 

If I understand you correctly I will only see a staking reward when my online computer actually generates a block. At that time I will then see all of the accumulated staking rewards from the last time my computer generated a block. So, if I don’t generate a block, will a rescan of the blockchain show previously unseen staking rewards? That would explain the rewards seen in December and January. 

 

I am confused because when I hover over the green staking icon the last line is “Expected to earn reward once every 16 hours”. As I explained in my previous email I was having a problem keeping the wallet on line (old quad core HP desktop running Win 10). The latest software release won’t run on it without crashing the system.

 

The wallet has been running fine on a Lenovo DT that I bought a year ago. Although, the Lenovo completely locks up now and then and has to be powered down to get out of it. I suspect it is due to one of the crypto wallets I have running on it (Reddcoin core, Neblio core, NoLimitCoin2 core) or perhaps a conflict between them. I know the problem didn’t start until after I put them on the Lenovo, but then it could have been a windows update interfering with the wallets. Who knows? 

 

I’ve tried running the wallets individually but, the Lenovo didn’t lock up. I suppose I could use separate PC’s for each wallet but, its not worth the hassle.

 

Anyway, thanks again for the help. I’ll keep the Reddcoin wallet online and wait on an update. If I don’t see an update before April I will register another issue on GitHub.

 

Todd

**Comment by reddink** [comment]
Thanks
Closing for now.
You can re-open if the issue is not resolved
